private/rules/artifact.bzl

```diff
@@ -1,19 +1,22 @@
 load("//:specs.bzl", "parse")
 load("//private:constants.bzl", "DEFAULT_REPOSITORY_NAME")
+load("//private:coursier_utilities.bzl", "get_simple_coord")
 
 def artifact(a, repository_name = DEFAULT_REPOSITORY_NAME):
-    artifact_obj = _parse_artifact_str(a) if type(a) == "string" else a
-    return "@%s//:%s" % (repository_name, _escape(artifact_obj["group"] + ":" + artifact_obj["artifact"]))
+    artifact_str = _make_artifact_str(a) if type(a) != "string" else _ensure_version(a)
+    return "@%s//:%s" % (repository_name, _escape(get_simple_coord(artifact_str)))
 
 def maven_artifact(a):
     return artifact(a, repository_name = DEFAULT_REPOSITORY_NAME)
 
 def _escape(string):
     return string.replace(".", "_").replace("-", "_").replace(":", "_")
 
-def _parse_artifact_str(artifact_str):
-    pieces = artifact_str.split(":")
-    if len(pieces) == 2:
-        return {"group": pieces[0], "artifact": pieces[1]}
-    else:
-        return parse.parse_maven_coordinate(artifact_str)
+def _ensure_version(a):
+    # fake :version part is due to get_simple_coord not supporting unversioned coords
+    return a if len(a.split(":")) > 2 else (a + ":fake_version")
```

Comment:
ugly hack, want to reuse `get_simple_coord` logic with support for type and classifier, but it doesn't support unversioned artifacts, also only works on strings

Reply:
I don't understand why you need all of this?  Is it so you can keep the one call to `get_simple_coord` ?  Why can't this logic be put in `get_simple_coord` instead?

Reply:
it can, but it may cause breaking changes to other use sites of that function

Reply:
did a quick look, looks like this should be safe so moved there now